includes/starter_content/class-starter-content-wordpress.php

```diff
@@ -0,0 +1,250 @@
+				return [
+					'title'          => $post_data['title']['rendered'],
+					'date'           => $post_data['date'],
+					'content'        => $post_data['content']['rendered'],
+					'featured_image' => $post_data['featured_media'],
+					'site_url'       => $site_url,
+				];
+			},
+			$response_json 
+		);
+
+		update_option( self::$existing_content_raw_data_option, $relevant_info );
+		return true;
+	}
+
+	/**
+	 * Create the Xth starter post.
+	 *
+	 * @param int $post_index Number of post to create (1st, second, etc.).
+	 * @return int ID of existing post if already created or ID of created post.
+	 */
+	public static function create_post( $post_index ) {
+		$existing_post_id = self::get_starter_post( $post_index );
+		if ( $existing_post_id ) {
+			return $existing_post_id;
+		}
+
+		if ( ! function_exists( 'wp_insert_post' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/post.php';
+		}
+
+		$existing_content_raw_data = get_option( self::$existing_content_raw_data_option, [] );
+		if ( ! $existing_content_raw_data ) {
+			return false;
+		}
+
+		if ( empty( $existing_content_raw_data[ $post_index ] ) ) {
+			return false;
+		}
+
+		$raw_post_data = $existing_content_raw_data[ $post_index ];
+		$post_data     = [
+			'post_title'   => sanitize_text_field( $raw_post_data['title'] ),
+			'post_status'  => 'publish',
+			'post_content' => wp_kses_post( $raw_post_data['content'] ),
+			'post_date'    => sanitize_text_field( $raw_post_data['date'] ),
+		];
+		$post_id       = wp_insert_post( $post_data );
+		if ( ! $post_id ) {
+			return false;
+		}
+
+		if ( ! empty( $raw_post_data['featured_image'] ) && ! empty( $raw_post_data['site_url'] ) ) {
+			$featured_image_id = (int) $raw_post_data['featured_image'];
+			if ( $featured_image_id ) {
+				$media_endpoint = trailingslashit( esc_url_raw( $raw_post_data['site_url'] ) ) . 'wp-json/wp/v2/media/' . $featured_image_id;
+				$media_response = wp_remote_get( $media_endpoint, [ 'timeout' => 60 ] ); //phpcs:ignore
+				if ( ! is_wp_error( $media_response ) && 200 === wp_remote_retrieve_response_code( $media_response ) ) {
+					$media_response_body = json_decode( wp_remote_retrieve_body( $media_response ), true );
+					$media_src           = $media_response_body['source_url'];
+
+					if ( ! function_exists( 'media_sideload_image' ) ) {
+						require_once ABSPATH . 'wp-admin/includes/media.php';
+						require_once ABSPATH . 'wp-admin/includes/file.php';
+						require_once ABSPATH . 'wp-admin/includes/image.php';
+					}
+
+					$image_id = media_sideload_image( $media_src, $post_id, null, 'id' );
+					if ( is_numeric( $image_id ) ) {
+						set_post_thumbnail( $post_id, $image_id );
+					}
+				}
+			}
+		}
+
+		self::mark_starter_post( $post_id, $post_index );
+
+		return $post_id;
+	}
+
+	/**
+	 * Create the homepage.
+	 *
+	 * @return post_id ID of existing homepage or ID of created homepage.
+	 */
+	public static function create_homepage() {
```

Comment:
Suggestion: this method is same for both approaches, and likely will stay the same for any other approach. It could be moved to the parent class.

Reply:
Oh wait, it's not the same. With the generated approach, some assumptions can be made about categories, which makes it different than in the import approach.